### PR TITLE
Add support for GL_NV_shader_invocation_reorder.

### PR DESCRIPTION
**Pull request by alelenv**
GLSL spec : https://github.com/KhronosGroup/GLSL/pull/196
SPV spec : https://github.com/KhronosGroup/GLSL/pull/196
Headers : https://github.com/KhronosGroup/SPIRV-Headers/pull/300

### CODE REVIEW

**Comment by alelenv** [comment]
@greg-lunarg or @johnkslang : Headers and spec are merged. Waiting for tools to be merged so I can flip the known_good

Can you start reviewing in the meantime?

Thanks

**Comment by alelenv** [comment]
Repinging @greg-lunarg and @johnkslang 

All bots are green
Please start reviewing

Thanks

**Comment by dgkoch** [comment]
@jeremy-lunarg can you help review / merge this?

**Comment by greg-lunarg** [comment]
Sorry for the delay. Looking now.

**Review by greg-lunarg — CHANGES_REQUESTED** [review]
Please fix Android-ndk build issue.

File: SPIRV/GlslangToSpv.cpp [hunk 1]
```diff
@@ -8947,11 +9261,14 @@ spv::Id TGlslangToSpvTraverser::getSymbolId(const glslang::TIntermSymbol* symbol
...
-        if (!(glslangIntermediate->isRayTracingStage() && glslangIntermediate->IsRequestedExtension(glslang::E_GL_EXT_ray_tracing)
+        if (!(glslangIntermediate->isRayTracingStage() &&
+              (glslangIntermediate->IsRequestedExtension(glslang::E_GL_EXT_ray_tracing) ||
+               glslangIntermediate->IsRequestedExtension(glslang::E_GL_NV_shader_invocation_reorder))
               && (builder.getStorageClass(id) == spv::StorageClassRayPayloadKHR ||
                   builder.getStorageClass(id) == spv::StorageClassIncomingRayPayloadKHR ||
                   builder.getStorageClass(id) == spv::StorageClassCallableDataKHR ||
-                  builder.getStorageClass(id) == spv::StorageClassIncomingCallableDataKHR))) {
+                  builder.getStorageClass(id) == spv::StorageClassIncomingCallableDataKHR) ||
+                  builder.getStorageClass(id) == spv::StorageClassHitObjectAttributeNV)) {
```
[the remark this answers is not in the document]
Comment (greg-lunarg) [reply SPIRV/GlslangToSpv.cpp:9271]: It looks like this will also fix the clang-ubuntu build issue.

**Review by greg-lunarg — APPROVED** [review]
LGTM. Thanks for the code!

**Pull request merged**